survey_certification_sending/models/survey_user_input.py

```diff
@@ -0,0 +1,55 @@
+# Copyright 2025 Tecnativa - Pilar Vargas
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import _, fields, models
+
+
+class SurveyUserInput(models.Model):
+    _inherit = "survey.user_input"
+
+    certification_sent = fields.Boolean(
+        default=False, help="It indicates if the certification has been sent."
+    )
+
+    def _mark_done(self):
+        for user_input in self:
+            if (
+                not user_input.survey_id.skip_certification_email
+                and user_input.survey_id.certification
+                and user_input.scoring_success
+            ):
+                if (
+                    user_input.survey_id.certification_mail_template_id
+                    and not user_input.test_entry
+                ):
+                    user_input.certification_sent = True
+        # Mark as completed and send email only if not deactivated.
+        return super(
+            SurveyUserInput,
+            self.with_context(
+                skip_certification_email=self.mapped("survey_id").filtered(
+                    lambda s: s.skip_certification_email
+                )
+            ),
+        )._mark_done()
```

Comment:
You shouldn't avoid the survey completion, just the mail sending:


https://github.com/OCA/OCB/blob/eab555dc127b3a60a0f2b019c39abfbb3b078ff5/addons/survey/models/survey_user_input.py#L219

Reply:
I have made the change to inject the correct ids into the context instead of the flag since it can lead to error if at some point a set of records with different values for that flag is treated.